src/parser/uniterm.rs

```diff
@@ -167,6 +167,40 @@ impl UniRecord {
     /// doesn't support the field path syntax: `{foo.bar.baz : Type}.into_type_strict()` returns an
     /// `Err`.
     pub fn into_type_strict(self) -> Result<Types, InvalidRecordTypeError> {
+        fn term_to_record_rows(
+            id: Ident,
+            rt: RichTerm,
+            tail: RecordRows,
+        ) -> Result<RecordRows, InvalidRecordTypeError> {
+            // At parsing stage, all `Rc`s must be 1-counted. We can thus call
+            // `into_owned()` without risking to actually clone anything.
+            match rt.term.into_owned() {
```

Comment:
Not a blocker, but is there some way we could make this call fail instead of silently cloning if the invariant in the comment ends up broken by someone? In the spirit of failing early.